File: Factory/SimpleFactory/Java/src/Pizza/Pizza.java
package eu.svetlin.design.patterns.factory.simplefactory.Pizza;

import java.util.ArrayList;
import java.util.List;

abstract public class Pizza {
    String name;
    String dough;
    String sauce;
    List<String> toppings = new ArrayList<>();

    public String getName() {
        return name;
    }

    public void prepare(){
        System.out.println("Preparing " + name);
    }

    public void bake(){
        System.out.println("Baking " + name);
    }

    public void cut(){
        System.out.println("Cutting " + name);
    }

    public void box(){
        System.out.println("Boxing " + name);
    }

    @Override
    public String toString() {
        return "Pizza { " +
                "name='" + name + '\'' +
                ", dough='" + dough + '\'' +
                ", sauce='" + sauce + '\'' +
                ", toppings=" + toppings +
                " }";
    }
}

File: Factory/SimpleFactory/TypeScript/src/Pizza/Pizza.ts
abstract class Pizza {
  name: string;
  dough: string;
  sauce: string;
  toppings: string[] = [];

  getName(): string {
    return this.name;
  }

  prepare() {
    console.log('Preparing ', this.name);
  }

  bake() {
    console.log('Baking ' + this.name);
  }

  cut() {
    console.log('Cutting ' + this.name);
  }

  box() {
    console.log('Boxing ' + this.name);
  }

  toString() {
    return `Pizza { name='${this.name}', dough='${this.dough}', sauce='${this.sauce}', toppings=${this.toppings} }`;
  }
}

export default Pizza;
